File: Add Code Here/Python/Binary_Search.py
def arr_input():
    n = int(input("Enter the size of the array: "))
    arr = list(map(int, input("Enter the elements of the array: ").split()))
    return arr

def binary_search(arr, key):
    left, right = 0, len(arr) - 1
    
    while left <= right:
        mid = left + (right - left) // 2
        
        if arr[mid] == key:
            return mid
        elif arr[mid] < key:
            left = mid + 1
        else:
            right = mid - 1
            
    return -1

def main():
    arr = arr_input()
    key = int(input("Enter the element to search: "))
    
    index = binary_search(arr, key)
    if index != -1:
        print(f"Element found at index: {index}")
    else:
        print("Element not found")

if __name__ == "__main__":
    main()

File: Java/BinarySearch.java
import java.util.*;

public class BinarySearch {
    public static int binarySearch(int[] arr, int target) {
        int left = 0;
        int right = arr.length - 1;

        while (left <= right) {
            int mid = left + (right - left) / 2;

            if (arr[mid] == target) {
                return mid; // Target found
            } else if (arr[mid] < target) {
                left = mid + 1; // Search the right half
            } else {
                right = mid - 1; // Search the left half
            }
        }
        return -1; // This means the target element  not found
    }

    public static void main(String[] args) {
      Scanner sc = new Scanner("System.in");
      System.out.println("Enter the number of elements");
      int n = sc.nextInt();
      int arr[] = new int[n];  // Try this as I/P :  {1, 3, 5, 7, 9, 11};
      for(int i = 0 ; i < arr.length ; i++){
        arr[i] = sc.nextInt();
      }
      System.out.println("Enter the target");
       int target = sc.nextInt();
       int result = binarySearch(arr, target);

        if (result == -1) {
            System.out.println("Element not present in the array.");
        } else {
            System.out.println("Element found at index: " + result);
        }
    }
}
